docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,38 +1,96 @@
 
 * TOC
 {:toc}
 
-### Building cloud-controller-manager for your cloud
+## Administration
 
-To build cloud-controller-manager for your cloud, follow these steps:
+### Requirements
 
-1. Write a cloudprovider that satisfies the [cloudprovider.Interface](https://git.k8s.io/kubernetes/pkg/cloudprovider/cloud.go).
-2. Link the cloudprovider to cloud-controller-manager.
+Every cloud has their own set of requirements for running their own cloud provider integration, it should not be too different from the requirements when running `kube-controller-manager`. As a general rule of thumb you'll need:
 
-The methods in [cloudprovider.Interface](https://git.k8s.io/kubernetes/pkg/cloudprovider/cloud.go) are self-explanatory. All of the
-[existing providers](https://git.k8s.io/kubernetes/pkg/cloudprovider/providers) satisfy this interface. If your cloud is already a part
-of the existing providers, you do not need to write a new provider; you can proceed directly with linking your cloud provider to the `cloud-controller-manager`.
-
-Once your code is ready, you must import that code into `cloud-controller-manager`. See the [rancher cloud sample](https://github.com/rancher/rancher-cloud-controller-manager) for a reference example. The import step in the sample is the only step required to link your cloud provider to the `cloud-controller-manager`.
+* cloud authentication/authorization: your cloud may require a token or IAM rules to allow access to their APIs
+* kubernetes authentication/authorization: cloud-controller-manager may need RBAC rules set to speak to the kubernetes apiserver
+* high availabilty: like kube-controller-manager, you may want a high available setup for cloud controller manager using leader election (on by default).
 
 ### Running cloud-controller-manager
 
-To run `cloud-controller-manager`, add it to your existing Kubernetes cluster as a Master component. All other master components except `kube-controller-manager` can be run without any changes.
+Successfully running cloud-controller-manager requires some changes to your cluster configuration.
+
+* `kube-apiserver` and `kube-controller-manager` must not specify a `--cloud-provider` flag or must set it to `--cloud-provider=external`. This ensures that it does not run any cloud specific loops that would be run by cloud controller manager. In the future, this flag will be deprecated and removed.
+* `kubelet` must run with `--cloud-provider=external`. This is to ensure that the kubelet is aware that it must be initialized by the cloud controller manager before it is scheduled any work.
```

Comment:
MAY BE run with `--cloud-provider=external` if your cloud needs to initialize nodes before becoming schedulable.
If that's not required, don't set the flag

Reply:
I would consider `--cloud-provider=external` is a requirement  since `AddCloudNode` is a no-op otherwise https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L280-L284 and nodecontroller is a very large part of CCM